sdk/python/lib/pulumi/resource.py

```diff
@@ -898,7 +910,8 @@ def create_urn(
         if project is None:
             project = get_project()
 
-        parent_prefix = "urn:pulumi:" + stack + "::" + project + "::"
+        parent_prefix = Output.from_input("urn:pulumi:" + stack + "::" + project + "::")
 
-    return Output.all(parent_prefix, type_, name).apply(
-        lambda arr: arr[0] + arr[1] + "::" + arr[2])
+    all_args = [parent_prefix, type_, name]
+    # invariant http://mypy.readthedocs.io/en/latest/common_issues.html#variance
+    return Output.all(*all_args).apply(lambda arr: arr[0] + arr[1] + "::" + arr[2]) # type: ignore
```

Comment:
alternatively, could you wrap `type_` and `name` in `Output.from_input` as well, then do the `Outout.all`?